corehq/apps/accounting/subscription_changes.py

```diff
@@ -34,7 +35,18 @@
 class BaseModifySubscriptionHandler(object):
 
     def __init__(self, domain, new_plan_version, changed_privs, date_start=None):
-        self.domain = domain if isinstance(domain, Domain) else Domain.get_by_name(domain)
+
+        def get_domain(domain):
+            obj = Domain.get_by_name(domain)
+            if obj:
+                return obj
+            else:
+                # This could happen in when there is an issue with couch cluster
+                #   that makes the obj not available
+                time.sleep(5)
```

Comment:
There are lots of reasons why a domain may not be found, and it seems likely that most of them will not magically resolve themselves after waiting for 5 seconds. Generally it's better to fail fast and have the client retry later if we suspect there was an error with the couch cluster. Sleeps like this can make it very hard to write speedy tests (not to mention production code).

Is there any way to reliably tell if the couch cluster has an "issue" that would cause this problem? Might be better to trigger an informative error on detecting that condition.

Reply:
What about adding a soft assert here, so we can track the issue?

This code is run during new user registration, which is one of the worst places to fail fast from a business perspective.  I'd hesitate to accept failing fast in hopes of solving the root cause anytime soon because there was quite a bit of investigation in that ticket that didn't turn up any good leads.  That said, I acknowledge this is a band-aid and not a fix for the root issue.

Reply:
Sprinkling `time.sleep()` around to fix issues like this is generally a horrible practice that should be avoided at all cost. This type of coding will make future maintenance much more difficult because it introduces dependence on subtle timing issues.

Presumably the domain being retrieved here was recently saved (in this same request). Is it possible to pass the recently saved domain object in here so it's not necessary to load it from the database again?

Reply:
Agree with your concern, Daniel. 

The code path is too deep to pass domain (it could be doable but it's a long chain of passing), though the main concern is that the same issue would manifest in a conflict error (or some other error) when the domain-object is saved.